server/workflow/workflow_server.go

```diff
@@ -33,6 +35,21 @@ func NewWorkflowServer(instanceID string, offloadNodeStatusRepo sqldb.OffloadNod
 	}
 }
 
+func (s *workflowServer) setInstanceID(instanceID string, wf *v1alpha1.Workflow) {
+	if len(instanceID) > 0 || len(s.instanceID) > 0 {
```

Comment:
I think it's clearer if we check if strings are empty like so, otherwise they could be confused for slices.
```suggestion
	if instanceID != "" || s.instanceID != "" {
```